bittensor/commands/wallets.py

```diff
@@ -827,3 +828,163 @@ def check_config(config: "bittensor.config"):
+
+GRAPHQL_QUERY = """
+query ($first: Int!, $after: Cursor, $filter: TransferFilter, $order: [TransfersOrderBy!]!) {
+    transfers(first: $first, after: $after, filter: $filter, orderBy: $order) {
+        nodes {
+            id
+            from
+            to
+            amount
+            extrinsicId
+            blockNumber
+        }
+        pageInfo {
+            endCursor
+            hasNextPage
+            hasPreviousPage
+        }
+        totalCount
+    }
+}
+"""
+
+
+class GetWalletHistoryCommand:
+    """
+    Executes the 'history' command to fetch the latest transfers of the provided wallet on the Bittensor network.
+
+    This command provides a detailed view of the transfers carried out on the wallet.
+
+    Usage:
+    The command lists the latest transfers of the provided wallet, showing the From, To, Amount, Extrinsic Id and Block Number.
+
+    Optional arguments:
+    None. The command uses the wallet and subtensor configurations to fetch latest transfer data associated with a wallet.
+
+    Example usage:
+    >>> btcli wallet history
+
+    Note:
+    This command is essential for users to monitor their financial status on the Bittensor network.
+    It helps in fetching info on all the transfers so that user can easily tally and cross check the transactions.
+    """
+
+    @staticmethod
+    def run(cli):
+        r"""Check the transfer history of the provided wallet."""
+        wallet = bittensor.wallet(config=cli.config)
+        wallet_address = wallet.coldkey.ss58_address
+
+        # Fetch all transfers
+        transfers = get_wallet_transfers(wallet_address)
+
+        # Create output table
+        table = create_transfer_history_table(transfers)
+
+        bittensor.__console__.print(table)
+
+    @staticmethod
+    def add_args(parser: argparse.ArgumentParser):
+        history_parser = parser.add_parser(
+            "history",
+            help="""Fetch transfer history associated with the provided wallet""",
+        )
+        bittensor.wallet.add_args(history_parser)
+        bittensor.subtensor.add_args(history_parser)
+
+    @staticmethod
+    def check_config(config: "bittensor.config"):
+        if not config.is_set("wallet.name") and not config.no_prompt:
+            wallet_name = Prompt.ask("Enter wallet name", default=defaults.wallet.name)
+            config.wallet.name = str(wallet_name)
+
+
+def get_wallet_transfers(wallet_address) -> List[dict]:
+    """Get all transfers associated with the provided wallet address."""
+
+    variables = {
+        "first": 10,
+        "filter": {
+            "or": [
+                {"from": {"equalTo": wallet_address}},
+                {"to": {"equalTo": wallet_address}},
+            ]
+        },
+        "order": "BLOCK_NUMBER_DESC",
+    }
+
+    transfers = []
+
+    # Make the request with the provided query and variables until all transfers are fetched
+    while True:
```

Comment:
Generally weary of unbounded `while True` statements in python. This technically *should* terminate on the desired condition, but may want to explicitly put the exit condition in this line and/or specify some other conditions for exiting. For example, if there is significant txn volume, we may want to limit the output to `n` txns.

For example, adding a `max_history_length` that we exit upon reaching:

```python
max_txn_history = 1000 # this should be set in the config.
page_info = {"page_info": {"hasNextPage": True}} # for initial loop entry, will be filled by `transfer_data`
while len(transfers) < max_txn_history and page_info.get("hasNextPage") == True:
    ...
```

What do you think?

Reply:
Yeah, I think it will be a better approach. Will be fixing this in the next commit.

Reply:
You not need while. Change 10 to 1000 or more

Reply:
@alex0500 removed while and set max_txn to 1000.